src/components/Common/FilePreviewDialog.tsx

```diff
@@ -276,48 +306,136 @@ const FilePreviewDialog = (props: FilePreviewProps) => {
   };
 
   const handleTouchStart = (e: React.TouchEvent) => {
-    if (!file_state.isImage) return;
-    setDragState((prev) => ({
-      ...prev,
-      isDragging: true,
-      dragStart: {
-        x: e.touches[0].clientX - prev.position.x,
-        y: e.touches[0].clientY - prev.position.y,
-      },
-    }));
+    if (e.touches.length === 2) {
+      const touch1 = e.touches[0] as unknown as Touch;
+      const touch2 = e.touches[1] as unknown as Touch;
+      const distance = getDistance(touch1, touch2);
+      const center = getCenter(touch1, touch2);
+
+      setPinchState({
+        isPinching: true,
+        initialDistance: distance,
+        initialScale: file_state.isImage ? file_state.zoom / 4 : scale,
+        center,
+      });
+
+      setDragState((prev) => ({
+        ...prev,
+        isDragging: false,
```

Comment:
isn't there any package that we could use to achieve this instead of writing by ourselves? 🤔